### DISCUSSION TITLE
TAGs, supported?

### DISCUSSION THREAD

**Discussion by isharris1** [pymodbus-dev/pymodbus#1414] [Q&A] [answered]
Hi, thanks for support this, its useful for learning! And I am still learning! So thats why my question.

Does pymodbus support TAGs?  

quote  "
_In more advanced PLC designs, a more structured programming approach is taken with the use of “tags,” which are stored within the PLC’s memory in a tag database.
With a Tag Database, all function blocks, including contacts, coils, program variables (e.g., as a timer value called “TransmitterRFMuteTimer”), as well all other objects, are stored as variables with attributes such as initial value, float, string, integer, Boolean (on/off), ASCII text, discreet inputs and discreet outputs. While this may seem confusing at first-glance, it’s a superior approach for more complex designs but does require (as in structured programming languages) the designer to declare these variables “tags,” as well as the data type in advance of their use in the program._
"

Is that a thing I can do with pymodbus? I've not seen anything in the docs to suggest it is.

**Comment by janiversen** [comment]
you are asking the wrong question, the correct question is does the modbus protocol support tags, and the answer to that is no...and as a consequence neither do pymodbus.

I think you have taken the quote a bit out of context, the quote talks about how to use allocate data in the server (the memory model) which in pymodbus terms is the server. We have a simulator module that do exactly that for the basic types. Some more modern protocols use tags for communication, but they depend on relative strong connections, while a protocol like modbus can run on very slow connections.

**Reply by isharris1** [reply]
Thanks, still figuring this out.

So rather than 'request Motor_Speed'.  I need to (in my HMI) maintain my own point/tag list of: Motor_Speed is in register 40001.  So I then request register40001 and do something in the HMI to match that data from specific Modbus device, to specific display in HMI?

**Reply by janiversen** [reply, marked as answer]
I think you need to start by reading the modbus protocol definition. You are mixing server/client issues.

**Reply by janiversen** [reply]
Look e.g. at how modbus integrates into HomeAssistant.

**Reply by isharris1** [reply]
Oh, homeassistant is a good suggestion. thanks! hadnt thought of that (probably easier 'at home!')
